File: xopus/apps/src/main/webapp/Xopus-4.3.1/examples/howto/SaveToRuby/save.rb
####################################################################################### 
#                                                                                     #
# REQUIREMENTS AND CONFIGURATIONS:                                                    #
#                                                                                     #
# This is an example for saving incoming xml in Ruby on Rails                         #
# For this example is tested with the following configuration                         #
# 1) Apache 2.2                                                                       #
# 2) Ruby on Rails 186-25                                                             #
# 3) RubyGems                                                                         #
# 4) libxml-ruby (if you want to use xmldom)                                          #
# 5) Windows XP                                                                       #
#                                                                                     #
# NOTES                                                                               #
#                                                                                     #
# - replace MySaveController with your own save class                                 #
# - replace pathRoot with your own root to document                                   #
# - if you want to parse it into an xmldom first then you can use                     #
#   XmlDom = XML::Parser.string(data).parse                                           #
#                                                                                     #
#######################################################################################


class MySaveController < ApplicationController
  def index
    data   = @request.env["RAW_POST_DATA"]    
        @content_type = "application/xml"
        pathRoot = "C:/xxx"
        fileName = "document.xml"
        
        File.open("#{pathRoot}/#{fileName}","wb") do |file| 
            file.puts data
        end
        
    render_text "document saved"
  end
  
end

File: xopus/apps/src/main/webapp/Xopus-4.3.1/examples/howto/SaveToPython/save.py
####################################################################################### 
#                                                                                     #
# REQUIREMENTS AND CONFIGURATIONS:                                                    #
#                                                                                     #
# This is an example for saving incoming xml in Python                                #
# For this example is tested with the following configuration                         #
# 1) Apache 2.2                                                                       #
# 2) Python 2.5                                                                       #
# 3) ElementTree 1.2.6 (You can download it from : http://effbot.org/downloads/)      #
# 4) Windows XP                                                                       #
#                                                                                     #
# NOTES                                                                               #
#                                                                                     #
# - ET.XLM()                                                                          #
#   creates an xmlElement using a string as input                                     #
#                                                                                     #
# - ET.parse()                                                                        #
#   use this if you want to load from a file                                          #
#                                                                                     #
# - Somehow the rawData in python is always UNICODE.                                  #
#   So don't forget to encode it back to utf8 when you work with xml                  #
#                                                                                     #
# - XOPUS always outputs valid XML. You may wonder why we still want to               #
#   parse the incoming xml back into a XmlDom first before saving.                    #
#   Here is the reason.                                                               #
#                                                                                     #
#   If you try to write the incoming string into a file directly after calling        #
#   rawData.encode() it will add an extra newline for every newline in the string.    #
#   Somehow there is a bug in the .encode. In order to get rid of those filthy        #
#   newlines we parse it into a xmltree first before we write it back to a file       #
#                                                                                     #
#                                                                                     #
#######################################################################################

from mod_python import apache
import sys
import xml.etree.ElementTree as ET

def XML(text):
    parser = ET.XML(text)
    return ET.ElementTree(parser)

def handler(req):
  rawData = ""
  for line in req.readlines(): 
     rawData += line 
  
  rawData.encode("utf8")  

  XmlDom = XML(rawData)
  
  XmlDom.write("document.xml","utf-8")    
  
  req.write("file is saved!")
  return apache.OK